### PR TITLE
[Fix] Extracted hardcoded text and fixed some French grammar

### LINKED ISSUE #151: Extract hardcoded text
**Description**
Need to extract hardcoded strings for them to be translated into french

### PR DESCRIPTION
**Pull request by BenJeau**
A lot of the changes are related to French grammar, but I extracted some hardcoded text in some components. Please let me know if I'm doing it correctly (not sure if I should use `intl.formatMessage` or `FormattedMessage`) and if the `react-intl` - injectIntl HOC should be used in Views

closes #151

### CODE REVIEW

**Comment by dynamic11** [comment]
> A lot of the changes are related to French grammar, but I extracted some hardcoded text in some components. Please let me know if I'm doing it correctly (not sure if I should use `intl.formatMessage` or `FormattedMessage`) and if the `react-intl` - injectIntl HOC should be used in Views
> 
> closes #151

As much as we can we use FormattedMessage because it helps us avoid using injectIntl. Some places you will need intl.formatMessage because it returns a string.

File: services/frontend-v3/src/components/mentorshipCard/MentorshipView.jsx [hunk 1]
```diff
@@ -39,7 +40,7 @@ function MentorshipView(props) {
       return (
         <Empty
           image={Empty.PRESENTED_IMAGE_SIMPLE}
-          description="no mentorship skills provided"
+          description={props.intl.formatMessage({ id: "profile.mentorship.empty" })}
```
Comment (dynamic11) [inline services/frontend-v3/src/components/mentorshipCard/MentorshipView.jsx]: use FormatedMessage in cases like this

File: services/frontend-v3/src/components/layouts/editProfileLayout/EditProfileLayoutView.jsx [hunk 2]
```diff
@@ -2,7 +2,7 @@ import React from "react";
 import { useHistory } from "react-router-dom";
 import { PageHeader, Menu } from "antd";
 import { RightOutlined } from "@ant-design/icons";
-import { FormattedMessage } from "react-intl";
+import { FormattedMessage, injectIntl } from "react-intl";
```
Comment (dynamic11) [inline services/frontend-v3/src/components/layouts/editProfileLayout/EditProfileLayoutView.jsx:5]: **Just a comment:**
I was wondering if it is bad to have both in the same file and found this:
[github issue comment](https://github.com/formatjs/formatjs/issues/942#issuecomment-345994000)

seems like its ok but let me know what you interpret from those comments

Comment (BenJeau) [reply services/frontend-v3/src/components/layouts/editProfileLayout/EditProfileLayoutView.jsx:5]: I think its fine, but we could just use `FormattedMessage` (acc to the issue) and get the string like this, but not sure if it complicates the code

```jsx
<FormattedMessage id="language.code" defaultMessage="en">
	{ locale => (<PageHeader
			style={{
				padding: "0 0 15px 7px",
   				textTransform: "capitalize",
				textTransform: locale === "en" ? "capitalize" : "",
			}}
			title={<FormattedMessage id="edit.profile" />}
			/>)}
</FormattedMessage>
```

Comment (dynamic11) [reply services/frontend-v3/src/components/layouts/editProfileLayout/EditProfileLayoutView.jsx:5]: I dont find that very clear tbh. so maybe we just stick to what we have

**Review by dynamic11 — APPROVED** [review]

**Pull request merged**